tests/Utils/UtilsTest.php

```diff
@@ -149,4 +156,154 @@ public function testCleanStringMethod($text, $expected, $errorMessage): void
+                'isCategoryPage' => false,
+                'expectedFindologicActive' => false,
+                'expectedSmartSuggestActive' => false,
+            ],
+            'Plugin is active' => [
+                'isActive' => true,
+                'isActiveOnCategory' => true,
+                'isDirectIntegration' => false,
+                'isStagingShop' => false,
+                'isStagingSession' => false,
+                'isCategoryPage' => false,
+                'expectedFindologicActive' => true,
+                'expectedSmartSuggestActive' => true,
+            ],
+            'Plugin is active but staging is enabled' => [
+                'isActive' => true,
+                'isActiveOnCategory' => true,
+                'isDirectIntegration' => false,
+                'isStagingShop' => true,
+                'isStagingSession' => false,
+                'isCategoryPage' => false,
+                'expectedFindologicActive' => false,
+                'expectedSmartSuggestActive' => false,
+            ],
+            'Plugin is active but staging is enabled and findologic=on is set' => [
+                'isActive' => true,
+                'isActiveOnCategory' => true,
+                'isDirectIntegration' => false,
+                'isStagingShop' => true,
+                'isStagingSession' => true,
+                'isCategoryPage' => false,
+                'expectedFindologicActive' => true,
+                'expectedSmartSuggestActive' => true,
+            ],
+            'Plugin is active and direct integration' => [
+                'isActive' => true,
+                'isActiveOnCategory' => true,
+                'isDirectIntegration' => true,
+                'isStagingShop' => false,
+                'isStagingSession' => false,
+                'isCategoryPage' => false,
+                'expectedFindologicActive' => false,
+                'expectedSmartSuggestActive' => true,
+            ],
+            'Plugin is active and direct integration with staging' => [
+                'isActive' => true,
+                'isActiveOnCategory' => true,
+                'isDirectIntegration' => true,
+                'isStagingShop' => true,
+                'isStagingSession' => true,
+                'isCategoryPage' => false,
+                'expectedFindologicActive' => false,
+                'expectedSmartSuggestActive' => true,
+            ],
+            'Plugin is active and direct integration with staging but without findologic=on' => [
+                'isActive' => true,
+                'isActiveOnCategory' => true,
+                'isDirectIntegration' => true,
+                'isStagingShop' => true,
+                'isStagingSession' => false,
+                'isCategoryPage' => false,
+                'expectedFindologicActive' => false,
+                'expectedSmartSuggestActive' => true, // Smart Suggest handles findologic=on in case of DI.
```

Comment:
> // Smart Suggest handles findologic=on in case of DI.

What does that mean? 
In case Staging shop but without the flag, I would assume that this should be `false`

Reply:
If it is Direct Integration, we will let the Direct Integration decide. It is not the job of the plugin to decide weither to enable direct integration or not.

Reply:
But DI should also detect that findologic should be inactive with this setting. How can this still be `true`? 
The property is by default false.

Reply:
This is also the same for Shopware 5 by the way.

Reply:
for clarification, which code is setting it to true?

Reply:
> But DI should also detect that findologic should be inactive with this setting. How can this still be true?
The property is by default false.

We explicitly check if it is Direct Integration. In this case, we will always show the Smart Suggest, no matter what, because we shouldn't care about it.

![image](https://user-images.githubusercontent.com/20555112/97603831-1a1df500-1a0d-11eb-9074-0dd3dfc53501.png)

Reply:
As discussed in call, this can be resolved.